script.js

```diff
@@ -1,4 +1,197 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+//==================== Create Deck ========================
+var makeDeck = function () {
+    var deck = [];
+  
+    var suits = ['hearts', 'diamonds', 'clubs', 'spades'];
+  
+    var suitIndex = 0;
+    while (suitIndex < suits.length) {
+      var currentSuit = suits[suitIndex];
+      var rankCounter = 1;
+      while (rankCounter <= 13) {
+        var cardName = rankCounter;
+  
+        if (cardName == 1) {
+          cardName = 'ace';
+        } else if (cardName == 11) {
+          cardName = 'jack';
+        } else if (cardName == 12) {
+          cardName = 'queen';
+        } else if (cardName == 13) {
+          cardName = 'king';
+        }
+//================= Single Card Variable ======================
+        var card = {
+          name: cardName,
+          suit: currentSuit,
+          rank: rankCounter,
+        };
+//================ Card To Deck ====================== 
+        deck.push(card);
+  
+        rankCounter = rankCounter + 1;
+      }
+      suitIndex = suitIndex + 1;
+    }
+  
+    return deck;
+  };
+  
+  var getRandomIndex = function (size) {
+    return Math.floor(Math.random() * size);
+  };
+//================= Shuffle Card ========================  
+  var shuffleCards = function (cards) {
+    var index = 0;
+  
+    while (index < cards.length) {
+      var randomIndex = getRandomIndex(cards.length);
+  
+      var currentItem = cards[index];
+  
+      var randomItem = cards[randomIndex];
+  
+      cards[index] = randomItem;
+      cards[randomIndex] = currentItem;
+  
+      index = index + 1;
+    }
+  
+    return cards;
+  };
+//================= System work =====================  
+  var deck = shuffleCards(makeDeck());
+  
+  var maxNum = 21;
+  var hitNum = 16;
+  var moveStand = false;
+  var gameOver = false;
+  var gameMode = "please enter player's name"
+  var playerName = ""
```

Comment:
It will be good to create all the possible game modes here as variables so it's easier to tell what game modes your game has.